### PR TITLE
bug fix of prevention of zero target optimization values by the evil atof()

### PR DESCRIPTION
**Pull request by psi-rking**
## Description
Fixes bug reported [here](http://forum.psicode.org/t/fixed-dihedral-optimization-error/323).  I was sloppy and apparently no one had tried a 0 degree fixed angle before.

* **User-Facing for Release Notes**
  - [ ] Fixes bug preventing fixed (target) values of exactly zero.

## Status
Passed HOOH at 0 degrees.  Still need to run other psi standard tests. 
- [ ]

### CODE REVIEW

**Comment by loriab** [comment]
Could `std::string` operations serve the purpose here? We had some `char *` issues with pybind11 and although there's no direct interface in optking, there was talk of moving from `char *`s to `std::string`s for readability.

**Comment by psi-rking** [comment]
OK, I pushed a stringstream version.

File: psi4/src/psi4/optking/molecule_read_coords.cc [hunk 1]
```diff
@@ -567,12 +567,24 @@ bool stoi(string s, int *a) {
 
 // convert string to float
 bool stof(string s, double *val) {
-  double i = atof(s.c_str());
-  if (i!=0) {
-    *val = i;
+   double d;
+   std::istringstream iss(s);
+   iss >> d;
+   if (iss.bad() || iss.fail())
```
Comment (fevangelista) [inline psi4/src/psi4/optking/molecule_read_coords.cc]: Just out of curiosity: wouldn't std::stod() be more convenient/compact than the istringstream solution?

**Comment by psi-rking** [comment]
I incorporated Francesco's suggestion of std::stod().  It worked fine in my tests.  Only for a hand-edited intco.dat would these checks ever matter.  (I discovered that if the user tries a string like "hello" for the value, it doesn't even pass through our initial psithon syntax code for some reason.)  The important thing is that we allow zero values.  BTW, I wonder why I can't see Francesco's comment on github's website (https://github.com/psi4/psi4/pull/535).  Are there private comments?

**Comment by loriab** [comment]
There aren't private comments, but GH does hide comments that refer to outdated code. If you click the "Show outdated" above, do you see what you wanted?

**Comment by loriab** [comment]
lgtm

**Comment by andysim** [comment]
lgtm

**Comment by dgasmith** [comment]
LGTM

**Pull request merged**